DartFileType.js

```diff
@@ -25,8 +25,8 @@ var Utils = require("loctool/lib/utils.js")
 var ResourceString = require("loctool/lib/ResourceString.js");
 
 var DartFileType = function(project) {
-    this.type = "dart";
-    this.datatype = "dart";
+    this.type = "x-dart";
```

Comment:
For matching with xliff files. As following the xliff file spec, user-defined values must start with an `x-` prefix. so the datatype in dart files in xliff is x-dart. so I changed it like this because the data must be the same when looking for a translation.